### PR TITLE
[F#] Add strings Concept Exercise

### LINKED ISSUE #758: [F#] Implement new Concept Exercise: string
This issue describes how to implement the `strings` concept exercise for the F# track.

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Strings in [F#][docs.microsoft.com-string].

## Things to teach

- The existence of the `string` type.
- How to create a string.
- The existence of `string` methods and `String` module function
- Basic string methods (like finding the index of a character at a position, or returning a part the string).
- Basic string formatting (using `+` and `sprintf`).
- Strings are immutable.

## Things not to teach

- Strings can be enumerated.
- String slicing.
- Triple-quoted strings.
- Double-backtick strings.
- Advanced string formatting (combining types e.g.).
- Using standard or custom format strings.
- Type inference.
- Memory and performance characteristics.

## Concepts

The Concepts this exercise unlocks are:

- `strings-basic`: know of the existence of the `string` type; know of some basic functions (like looking up a character at a position, or slicing the string); know how to do basic string concatenation.

## Prerequisites

There should be no prerequisites.

### Hints

- [String class][docs.microsoft.com-string]
- [String module][docs.microsoft.com-string-module]
- [String concatenation][docs.microsoft.com-string-concatenation]

### After

- [Strings reference](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/strings): includes details on verbatim strings, triple-quoted strings and slicing.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement the concept exercise, the following files must be created:

<pre>
languages
└── fsharp
    └── exercises
        └── concept
            └── strings
                ├── .docs
                |   ├── instructions.md
                |   ├── introduction.md
                |   ├── hints.md
                |   └── after.md
                ├── .meta
                |   |── design.md
                |   └── Example.fs
                ├── Strings.fs
                ├── Strings.fsproj
                └── StringsTest.fs
</pre>

## Step 1: add .docs/introduction.md

This file contains an introduction to the concept. It should be explicit about what the exercise teaches and maybe provide a brief introduction to the concepts, but not give away so much that the user doesn't have to do any work to solve the exercise.

## Step 2: add .docs/instructions.md

This file contains instructions for the exercise. It should explicitly explain what the user needs to do (define a method with the signature `X(...)` that takes an A and returns a Z), and provide at least one example usage of that function. If there are multiple tasks within the exercise, it should provide an example of each.

## Step 3: add .docs/hints.md

If the user gets stuck, we will allow them to click a button requesting a hint, which shows this file. We will softly discourage them using it. The file should contain both general and task-specific "hints". These hints should be enough to unblock almost any student.

## Step 4: add .docs/after.md

Once the user completes the exercise they will be shown this file, which gives them any bonus information or further reading about the concept taught.

These files are also all described in the [concept exercises document][docs-concept-exercises].

## Step 5: update languages/fsharp/config.json

An entry should be added to the track's `config.json` file for the new concept exercise:

```json
{
  ...
  "exercises": {
    "concept": [
      ...
      {
        "slug": "strings",
        "uuid": "9c2aad8a-53ee-11ea-8d77-2e728ce88125",
        "concepts": ["strings-basic"],
        "prerequisites": []
      }
    ]
  }
}
```

## Step 6: adding track-specific files

These files are specific to the F# track:

- `<NAME>.fs`. the stub implementation file, which is the starting point for students to work on the exercise.
- `<NAME>.fsproj`: the F# project file.
- `<NAME>Test.fs`: the test suite.
- `.meta/Example.fs`: an example implementation that passes all the tests. The project file should _not_ include this file.

## Step 7: update the general concept document

Add the exercise to the [concept's shared document's][reference] `## Implementations` section ([example](https://github.com/exercism/v3/blob/master/reference/types/string.md#implementations)).

## Step 8: updating list of implemented exercises

- Add the exercise to the [list of implemented exercises][implemented-exercises].

## Step 9: add .meta/design.md:

This file contains information on the exercise's design, which includes things like its goal, its teaching goals, what not to teach, and more ([example][meta-design]). This information can be extracted from this GitHub issue.

## Step 10: add reference documentation

This exercise could benefit from having a reference document written to introduce basic string concatenation in F#. See [this issue][string-concat-issue]

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[reference]: https://github.com/exercism/v3/blob/master/languages/fsharp/reference/README.md#reference-docs
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[docs.microsoft.com-string]: https://docs.microsoft.com/en-us/dotnet/api/system.string?view=netcore-3.1
[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[meta-design]: https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/enums-advanced/.meta/design.md
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[docs.microsoft.com-string-module]: https://msdn.microsoft.com/visualfsharpdocs/conceptual/core.string-module-%5bfsharp%5d
[docs.microsoft.com-string-concatenation]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/strings#string-operators
[string-concat-issue]: https://github.com/exercism/v3/issues/757
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes #758

### CODE REVIEW

File: languages/fsharp/exercises/concept/strings/.docs/after.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+The key thing to remember about F# strings is that they are immutable objects representing text as a sequence of Unicode characters (letters, digits, punctuation, etc.).
+
+Manipulating a string can be done by calling one of its [methods][methods] or [properties][properties], or using one of the functions in the [`String` module][string-module]. As string values can never change after having been defined, all string manipulation methods/functions will return a new string.
+
+A string is delimited by double quote (`"`) characters. Some special characters need [escaping][escaping] using the backslash (`\`) character. Strings can also be prefixed with the at (`@`) symbol, which makes it a [verbatim string][verbatim] that will ignore any escaped characters.
+
+Alternatively, strings can be defined using triple quotes (`"""`), which allows using a double quote in the string without escaping it. [Triple-quoted strings][triple-quoted] can also span multiple lines.
+
+Finally, concatenating strings can be done through the [`+` operator][plus-operator]. For any string formatting more complex than simple concatenation, the `sprintf` function is preferred.
```
Comment (robkeim) [inline languages/fsharp/exercises/concept/strings/.docs/after.md]: Do we want to reference `StringBuilder` here as the ideal way to concatenate multiple strings for performance reasons?

Comment (ErikSchierboom) [reply languages/fsharp/exercises/concept/strings/.docs/after.md]: I'd prefer to not mention it yet, as `StringBuilder` means knowing about state and it is mostly used for performance reasons, which I'd prefer not to bother the student with in this introductory exercise. I've added a note to the `design.md` file that mentions not wanting to teach the `StringBuilder` class.

**Comment by ErikSchierboom** [comment]
Thanks for the review(s) @robkeim!

File: languages/fsharp/exercises/concept/strings/.docs/after.md (hunk 1 above)
Comment (robkeim) [reply languages/fsharp/exercises/concept/strings/.docs/after.md]: Makes sense to me!

**Pull request merged**